packages/sync-actions/test/product-sync-variants.spec.js

```diff
@@ -345,6 +350,27 @@ describe('Actions', () => {
     ])
   })
 
+  it('should handle unsetting the key of a variant', () => {
+    const before = {
+      id: '123',
+      masterVariant: {
+        id: 1, key: 'v1', attributes: [{ name: 'foo', value: 'bar' }],
+      },
+    }
+
+    const now = {
+      id: '123',
+      masterVariant: {
+        key: '', attributes: [{ name: 'foo', value: 'bar' }],
+      },
+    }
+
+    const actions = productsSync.buildActions(now, before)
+    expect(actions).toEqual([
+      { action: 'setProductVariantKey', key: null, variantId: 1 },
```

Comment:
Nitpick: I think `null` should be fine, but usually we don't set the key at all when "unsetting" a field value.